src/NuGet.Core/NuGet.Configuration/PackageNamespaces/SearchTree.cs

```diff
@@ -24,7 +24,9 @@ internal SearchTree(PackageNamespacesConfiguration configuration)
             {
                 foreach (string namespaceId in namespacePerSource.Value)
                 {
-                    Add(namespacePerSource.Key, namespaceId);
+                    PackageSource packageSource = configuration.GetPackageSource(namespacePerSource.Key);
```

Comment:
Alternatively, you could consider changing `internal PackageNamespacesConfiguration(Dictionary<string, IReadOnlyList<string>> namespaces)` to something that contains a tuple or package source as the key.

Might make also make it more complicated. 
Something to consider.

Reply:
I tried it but it looks more complicated, for example `NuGet.Configuration` target net45 which doesn't have ValueTuple and also I have to make many more changes to `SearchTreeTest.cs`. This part is still internal and we can address it later without breaking api if it's really necessary.